### PR TITLE
Form validation login register

### LINKED ISSUE #35: Form validation for login and register

### LINKED ISSUE #85: Challenge solved badge is not correctly shown after submitting a correct flag
I think the function `isSolved` is better put at computed instead of methods?

### PR DESCRIPTION
**Pull request by EricLung**
Closes #35. Closes #85.

### CODE REVIEW

File: front-end/src/components/TheAppNavigationBar.vue [hunk 1]
```diff
@@ -72,7 +73,8 @@ export default {
   methods: {
     async logout () {
       await store.dispatch('sessions/logout')
-      store.commit('users/resetGetMe')
+      await store.commit('users/resetGetMe')
```
Comment (samueltangz) [inline front-end/src/components/TheAppNavigationBar.vue]: why do we need `async` in `resetGetMe`?

File: front-end/src/views/Challenge.vue [hunk 2]
```diff
@@ -126,6 +126,7 @@ export default {
   mounted () {},
   updated () {},
   destroyed () {
+    store.dispatch('users/getMe')
```
Comment (samueltangz) [inline front-end/src/views/Challenge.vue]: call `getMe` when users correctly submits a flag instead of leaving the challenge

Comment (EricLung) [reply front-end/src/views/Challenge.vue]: I try that but some wired bug appear: The success message not shown.

Comment (EricLung) [reply front-end/src/views/Challenge.vue]: `get` request will refresh the page and `submitFlag.done` is reset.

**Pull request merged**